include/nbl/builtin/hlsl/cpp_compat/vector.hlsl

```diff
@@ -62,33 +67,25 @@ glm::vec<N, bool> operator>=(const glm::vec<N, T>& lhs, const glm::vec<N, T>& rh
 {
     return glm::greaterThanEqual<N, T>(lhs, rhs);
 }
+}
+#endif
 
-// TODO[Przemek]: implement hlsl version of this function
-template<typename T>
-struct ValueType
+namespace nbl
 {
-    typedef T::value_type type;
-};
-
-#else
-
-// TODO[Przemek]: figure out if there is a better way..
-// TODO[Przemek]: implement for more types
-
-template<typename T>
-struct ValueType
-{
-    using type = void;
-};
-template<>
-struct ValueType<float3>
+namespace hlsl
 {
-    using type = float;
+template<typename V>
+struct scalar_type
+{
+    using type = void;
 };
 
-#endif
-
-}
+template<typename T, uint16_t N>
+struct scalar_type<vector<T,N> >
+{
+    using type = T;
+};
+}

```

Comment:
you want to create a `type_traits.hlsl` and slap that there along with specializations

Reply:
done